### ISSUE TITLE
Can't apply units to a range

### ISSUE THREAD

**Issue by mcabbott** [SymbolicML/DynamicQuantities.jl#157]
I think this should work:
```julia
julia> (1:0.25:4)u"inch"
ERROR: Cannot create an additive identity from `Type{<:Quantity}`, as the dimensions are unknown. Please use `zero(::Quantity)` instead.

julia> collect(1:0.25:4)u"inch"
13-element Vector{Quantity{Float64, Dimensions{FixedRational{Int32, 25200}}}}:
 0.025400000000000002 m
              0.03175 m
               0.0381 m
              0.04445 m
...
```
Came up in https://github.com/JuliaLang/julia/issues/56610, which is about `1u"inch":0.25u"inch":4u"inch"`. Constructing a range like that will (bugs aside) circumvent all the "do what I mean" floating-point magic of `0:0.1:1`. But making the range first & then applying units would mean the magic applies to the numbers as written.

**Comment by mcabbott** [comment]
There are special method for scalar * range, which try to produce more ranges, instead of the `AbstractVector` path which makes a `Vector`. One of these must call `zero(T)`.

**Comment by MilesCranmer** [comment]
Got it, thanks. I put a fix in #158, want to take a look?
